src/integrationtests/java/com/aws/greengrass/integrationtests/DiskSpoolIntegrationTest.java

```diff
@@ -142,15 +139,11 @@ void GIVEN_disk_spool_plugin_WHEN_operation_fails_with_database_corruption_THEN_
             f.writeBytes("Garbage");
         }
 
-        // Fail to add second message
-        SpoolMessage spoolMessage2 = SpoolMessage.builder().id(2L).request(request2).build();
-        assertThrows(IOException.class, () -> diskSpool.add(2L, spoolMessage2));
-
-        // Successfully add third Message
-        SpoolMessage spoolMessage3 = SpoolMessage.builder().id(3L).request(request3).build();
-        diskSpool.add(3L, spoolMessage3);
-        String readMessage3 = new String (diskSpool.getMessageById(3L).getRequest().getPayload(), StandardCharsets.UTF_8);
-        assertEquals(message3, readMessage3);
+        // Successfully add Second Message
+        SpoolMessage spoolMessage3 = SpoolMessage.builder().id(2L).request(request2).build();
+        diskSpool.add(2L, spoolMessage3);
+        String readMessage3 = new String (diskSpool.getMessageById(2L).getRequest().getPayload(), StandardCharsets.UTF_8);
+        assertEquals(message2, readMessage3);
```

Comment:
What's happening here, is `ensureConnection()` preemptively checking for corrupt/unhealthy databases . If that's the case do we still need `checkAndHandleCorruption`

Reply:
ensureConnection() will check connection exists or not if not creates new connection we using it for ensuring  connection before read or write operations to database like insert, remove, getspoolmessageid methods. But where checkAndHandleCorruption() will do if it finds database corrupted closes the connection, delete database then initializes new connection and setups the new database. Here in the above test when as db is corrupted checkAndHandleCorruption() insert  next message in the new database

Reply:
Yes, but my question is based on what's changed here in the test> 
Why does the second message add successfully after database file is altered with? Does it mean that `ensureConnection` is doing something similar to what `checkAndHandleCorruption` was meant to do?